modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/OnflightMessageTracker.java

```diff
@@ -878,4 +906,67 @@ public void dumpMessageStatusToFile(File fileToWrite) throws AndesException {
         }
     }
 
+    /**
+     * Concurrent List like implementation
+     *
+     * @param <T>
+     *         Type of list items
+     */
+    private static class ConcurrentTrackingList<T> {
```

Comment:
@pumudu88  @a5anka : please add more description for this data structure (what's the advantage of using it). and investigate weather this can be used else where as well.